[android] Clear local Maven repo of obsolete packages

sjchmiela opened the pull request:
# Why

Once https://github.com/expo/expo/pull/10069 lands, having contents of `android/maven` committed will be obsolete.

# How

- removed prebuilt packages from `android/maven`
  - we should be safe to remove `android-jsc` as
    1. Turtle installs workspace JS dependencies
    2. we have a reference to _local Maven repo_ at `node_modules/android-jsc` in root `build.gradle`
  - our prebuilt packages

# Test Plan

- [x] shell app builds
- [x] shell app is buildable by Turtle (https://staging.expo.io/dashboard/sjchmiela/builds/5cd9d671-0f1d-4047-a9bd-67e7d4c20262/logs)
- [x] we are still able to build AARs for the packages if we would like to debug the shell app workflow

tsapeta commented on tools/expotools/src/commands/AndroidBuildPackages.ts:
```diff
@@ -382,10 +390,19 @@ export default (program: any) => {
+        throw new Error('Could not infer SDK version, please run with `--sdkVersion SDK_VERSION`');
+      }
+
+      await action({ sdkVersion, ...options });
```
You probably wanted to do
```suggestion
      await action({ ...options, sdkVersion });
```
otherwise `options.sdkVersion` would still be `undefined` if not passed 😉

sjchmiela replied:
Fixed in https://github.com/expo/expo/pull/9983/commits/e97bcfee23efac69bc729c4631c84cf9ee5b6cbf! Good catch!

tsapeta reviewed (commented):
Looks good so far, thanks for making my idea real! 👍 In the future I think it might be easier to submit separate PR that removes such big number of files as it's hard to review it right now 😅

sjchmiela commented:
Yeah, I agree, that's why I prepared the Git commit history so meticulously so you can go through the changes step-by-step:
- [`android` changes](https://github.com/expo/expo/pull/9983/files/c9b45a2489c8f588354f9667f8283a1a0f7a8479)
- [`expotools` changes](https://github.com/expo/expo/pull/9983/files/c9b45a2489c8f588354f9667f8283a1a0f7a8479..07467486e8460f16dd5d001e3a396a0bfeaa9d60)
- [CI changes](https://github.com/expo/expo/pull/9983/files/07467486e8460f16dd5d001e3a396a0bfeaa9d60..4fe5ae0e15aa6a2b3b9468930c9ca0b6704b60ce)
- [guides changes](https://github.com/expo/expo/pull/9983/commits/5c5412ac4963d57665282e5e2af7a4c307f620f2)

sjchmiela commented:
Actually so meticulously some changes got dropped while rebasing. Fixing!

sjchmiela committed:
[android] Remove `org.webkit` from Android maven repo

We should be safe as we have a reference to `node_modules/android-jsc` in root `build.gradle` AND we install root JS dependencies in Turtle

sjchmiela committed:
[android] Remove obsolete prebuilt packages
